src/main/java/online/book/store/model/User.java

```diff
@@ -31,4 +44,46 @@ public class User {
     private String shippingAddress;
     @Column(nullable = false)
     private boolean isDeleted = false;
+    @ManyToMany(fetch = FetchType.EAGER)//TODO:EAGER is ok?
+    @JoinTable(name = "users_roles",
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id"))
+    private Set<Role> roles;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return roles.stream()
+                .map(role -> new SimpleGrantedAuthority(role.getName().name()))
+                .collect(Collectors.toSet());
+    }
```

Comment:
Good job, but I think you can simplify it.
Let's make `getAuthority()` method in Role.class that should return role.name()